File: lib/data/non/lua/audio.lua
non.audio = {}

local Gdx = NON_GDX
  
function non.audio.sound(path, type)
  return Gdx.audio:newSound(non.files.open(path, type))
end
  
function non.audio.music(path, type)
  return Gdx.audio:newMusic(non.files.open(path, type))
end
  
function non.audio.play(source)
  source:play()
end
  
function non.audio.pause(source)
  source:pause()
end
  
function non.audio.resume(source)
  source:resume()
end
  
function non.audio.stop(source)
  source:stop()
end
  
function non.audio.loop(source)
  source:loop()
end

File: lib/data/non/ruby/audio.rb
class Audio
  java_import 'com.badlogic.gdx.Gdx'
  java_import 'non.Non'
  
  def sound(path, type = :internal)
    Gdx.audio.newSound NON.files.open(path, type)
  end
  
  def music(path, type = :internal)
    Gdx.audio.newMusic NON.files.open(path, type)
  end
  
  def play(source)
    source.play
  end
  
  def pause(source)
    source.pause
  end
  
  def resume(source)
    source.resume
  end
  
  def stop(source)
    source.stop
  end
  
  def loop(source)
    source.loop
  end
end